api/src/services/roles.ts

```diff
@@ -146,7 +147,45 @@ export class RolesService extends ItemsService {
 		return;
 	}
 
-	override async updateOne(key: PrimaryKey, data: Record<string, any>, opts?: MutationOptions): Promise<PrimaryKey> {
+	private isIpAccessValid(value?: any[] | null): boolean {
+		if (!value || !Array.isArray(value)) return false;
+
+		if (value.length === 0) return true;
+
+		for (const ip of value) {
+			try {
+				getMatch(ip);
+			} catch {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	private assertValidIpAccess(partialItem: Partial<Item>): void {
+		if ('ip_access' in partialItem && !this.isIpAccessValid(partialItem['ip_access'])) {
+			throw new InvalidPayloadError({ reason: 'IP-Access contains an incorrect value' });
```

Comment:
I'd use "IP Access" (instead of "IP-Access) which is how the field appears in the Data Studio:
```suggestion
			throw new InvalidPayloadError({ reason: 'IP Access contains an incorrect value' });
```

Also wondering whether it would be helpful to throw a more precise error message:
```suggestion
			throw new InvalidPayloadError({ reason: 'IP Access must consist of IP addresses, IP ranges or CIDR blocks' });
```